### PR TITLE
Mutation social modifiers moved to JSON

### PR DESCRIPTION
**Pull request by RadHazard**
The mutations modifiers for lie/persuasion/intimidation checks are now stored in JSON rather than being hard-coded.  I kept all the bonuses and penalties exactly as they were for this PR, but I think we need to make a second pass to rebalance them and add them to more mutations.  (The one exception is that I removed the conflict between butterfly wings and flowers, since those are both post-threshold mutations and you can never have both at the same time)

### CODE REVIEW

File: src/character.cpp [hunk 1]
```diff
@@ -2550,6 +2550,45 @@ std::string Character::extended_description() const
...
+            if( !std::all_of( soc_mod.required_mutations.begin(), soc_mod.required_mutations.end(),
+                [this]( const trait_id &need ) {
+                    return has_trait( need );
+                } ) ) {
+                continue;
+            }
+
+            if ( type == "lie" ) {
+                mut_mod += soc_mod.lie;
+            } else if ( type == "persuade" ) {
```
Comment (Night-Pryanik) [inline src/character.cpp]: Using `else if` means that we can't get modifiers for several types, e.g. `lie` AND `persuade`, or I'm getting it wrong?

Also, astyle.

Comment (RadHazard) [reply src/character.cpp]: I can't think of any circumstances when you'd need to more than one.  I could change it to just return a struct of all the values at once, though.

Come to think of it, that would get rid of the ugly string types as well

File: data/json/mutations.json [hunk 2]
```diff
@@ -487,7 +490,12 @@
         "points" : 1,
         "description" : "You have no qualms about bending the truth, and have practically no tells.  Telling lies and otherwise bluffing will be much easier for you.",
         "starting_trait" : true,
-        "cancels" : ["TRUTHTELLER"]
+        "cancels" : ["TRUTHTELLER"],
+        "social_modifiers" : {
+            "//" : "The tail acts as a tell, preventing this trait from working",
```
Comment (Coolthulhu) [inline data/json/mutations.json]: That is very specific and weird.
Would be much better without this single special case and the whole `blocker_mutations` mechanic that needs to be maintained for this tiny case.

Comment (RadHazard) [reply data/json/mutations.json]: I kind of agree, I was just replicating the logic was already in the hardcoded version.  I can drop it and simplify if you think it makes sense.  On the other hand, the system might come in handy if social modifiers were applied to more mutations.

Thoughts?  I'm okay with going either way on this one

Comment (Coolthulhu) [reply data/json/mutations.json]: It looks like someone just added an `else` by accident. Most such weird interactions have explicit comments.

Comment (RadHazard) [reply data/json/mutations.json]: Fair enough.  I thought I remembered it being an intentional interaction, but my memory is probably just failing me.  Either way, I'll cut it out for now

Comment (kevingranade) [reply data/json/mutations.json]: I'm also fine with just removing this.

**Pull request merged**